cocos/network/SocketIO.cpp

```diff
@@ -57,32 +52,32 @@ class SocketIOPacketV10x;
 class SocketIOPacket
 {
 public:
-    typedef enum
+    enum class SocketIOVersion
     {
         V09x,
         V10x
-    }SocketIOVersion;
+    };
 
     SocketIOPacket();
     virtual ~SocketIOPacket();
-    void initWithType(std::string packetType);
+    void initWithType(const std::string& packetType);
     void initWithTypeIndex(int index);
 
-    std::string toString();
-    virtual int typeAsNumber();
-    std::string typeForIndex(int index);
+    std::string toString()const;
+    virtual int typeAsNumber()const;
+    const std::string& typeForIndex(int index)const;
```

Comment:
Why return `std::string&`? I think we don't want the developer to modify the return value of `typeForIndex`.